macros/src/ast_node.rs

```diff
@@ -0,0 +1,172 @@
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote, ToTokens};
+use std::collections::HashMap;
+use syn::{parse_quote, punctuated::Punctuated, Fields, Ident, ItemEnum, Token};
+
+pub struct State<'a> {
```

Comment:
can you put the relevant PR description into the commit message, to help `git blame`

Reply:
I'm sorry, I don't understand what you mean by this.

Reply:
It's often a good idea to explain why some decisions where made,
the traditional place to put such explanation is the commit message.
For example it might be good explain why `blanket_ref` and `NodeRef` exist.

See also https://github.com/git/git/blob/79bdd48716a4c455bdc8ffd91d57a18d5cd55baa/Documentation/SubmittingPatches#L110-L225

Reply:
Since I've pushed some additional commits since your initial review, should I do this as part of a second squash commit or what?